pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -424,3 +431,26 @@ func (p *Provider) DeleteLaunchTemplates(ctx context.Context, nodeClass *v1beta1
 	}
 	return nil
 }
+
+func (p *Provider) ResolveClusterCIDR(ctx context.Context) error {
+	if p.ClusterCIDR.Load() != nil {
+		return nil
+	}
+	out, err := p.eksapi.DescribeClusterWithContext(ctx, &eks.DescribeClusterInput{
+		Name: aws.String(options.FromContext(ctx).ClusterName),
+	})
+	if err != nil {
+		return cloudprovider.NewNodeClassNotReadyError(fmt.Errorf("resolving cluster CIDR, %w", err))
+	}
+	if ipv4CIDR := out.Cluster.KubernetesNetworkConfig.ServiceIpv4Cidr; ipv4CIDR != nil {
+		p.ClusterCIDR.Store(ipv4CIDR)
+		logging.FromContext(ctx).With("cluster-cidr", *ipv4CIDR).Debugf("discovered cluster CIDR")
+		return nil
+	}
+	if ipv6CIDR := out.Cluster.KubernetesNetworkConfig.ServiceIpv6Cidr; ipv6CIDR != nil {
+		logging.FromContext(ctx).With("cluster-cidr", *ipv6CIDR).Debugf("discovered cluster CIDR")
+		p.ClusterCIDR.Store(ipv6CIDR)
```

Comment:
nit: for parallelism would be nice to match the stanza above with Store() then logging.

Also thoughts on adding something like a "cidr-type" in the structured values too?
```suggestion
		p.ClusterCIDR.Store(ipv6CIDR)
		logging.FromContext(ctx).With("cluster-cidr", *ipv6CIDR, "cidr-type", const.IPV6).Debugf("discovered cluster CIDR")
```

Reply:
Leaning back to not including, the format is obviously different enough.